src/cli-sdk/src/commands/cache.ts

```diff
@@ -0,0 +1,306 @@
+import { error } from '@vltpkg/error-cause'
+import { CacheEntry, RegistryClient } from '@vltpkg/registry-client'
+import { Spec } from '@vltpkg/spec'
+import { mkdir, rm } from 'node:fs/promises'
+import prettyBytes from 'pretty-bytes'
+import type { LoadedConfig } from '../config/index.ts'
+import type { CommandUsageDefinition } from '../config/usage.ts'
+import { commandUsage } from '../config/usage.ts'
+import type { CommandFn, CommandUsage } from '../index.ts'
+import { stdout } from '../output.ts'
+import type { ViewOptions, Views } from '../view.ts'
+import { ViewClass } from '../view.ts'
+
+export type CacheMap = Record<
+  string,
+  ReturnType<CacheEntry['toJSON']>
+>
+
+export type CacheSubcommands = keyof (typeof usageDef)['subcommands']
+
+let view: CacheView
+export class CacheView extends ViewClass {
+  constructor(options: ViewOptions, conf: LoadedConfig) {
+    super(options, conf)
+    view = this
+  }
+  stdout(...args: unknown[]) {
+    stdout(...args)
+  }
+  error(e: unknown) {
+    throw e
+  }
+}
+
+export const views: Views<void | CacheMap> = {
+  human: CacheView,
+}
+
+const usageDef = {
+  command: 'cache',
+  usage: '<command> [flags]',
+  description: 'Work with vlt cache folders',
+
+  subcommands: {
+    add: {
+      usage: '<package-spec> [<package-spec>...]',
+      description:
+        'Resolve the referenced package identifiers and ensure they are cached',
+    },
+
+    ls: {
+      usage: '[<key>...]',
```

Comment:
what are `<key>` in this context? integrity values?

Reply:
No, the keys that are printed out if you do `vlt cache ls` without any arguments. (Most of time, these will be URLs.)